modules/stochastic_tools/src/tests/TestDistributionPostprocessor.C

```diff
@@ -21,37 +14,30 @@ validParams<TestDistributionPostprocessor>()
   InputParameters params = validParams<GeneralPostprocessor>();
   params.addRequiredParam<DistributionName>(
       "distribution", "The distribution which supplies the postprocessor value.");
-  params.addParam<Real>(
-      "cdf_value", 0.5, "A cdf value that used to determine the value of sampled parameter");
-  params.addParam<Real>("sample_value",
-                        0.5,
-                        "A given sampled value that can be used to determine "
-                        "the probability and cumulative probability.");
+  params.addRequiredParam<Real>(
+      "value", "A value to pass to the cdf, pdf, or quantile function of the given distribution.");
 
+  MooseEnum method("cdf pdf quantile");
+  params.addParam<MooseEnum>("method", method, "The distribution method to call.");
   return params;
 }
 
 TestDistributionPostprocessor::TestDistributionPostprocessor(const InputParameters & parameters)
   : GeneralPostprocessor(parameters),
     _distribution(getDistribution("distribution")),
-    _cdf_value(getParam<Real>("cdf_value")),
-    _sample_value(getParam<Real>("sample_value"))
-{
-}
-
-void
-TestDistributionPostprocessor::initialize()
-{
-}
-
-void
-TestDistributionPostprocessor::execute()
+    _value(getParam<Real>("value")),
+    _distribution_method(getParam<MooseEnum>("method"))
 {
 }
 
 PostprocessorValue
 TestDistributionPostprocessor::getValue()
 {
-  return _distribution.cdf(_sample_value) * _distribution.pdf(_sample_value) *
-         _distribution.inverseCdf(_cdf_value) * _distribution.getRandomNumber();
+  if (_distribution_method == "pdf")
```

Comment:
These string comparisons might destroy you in practice on a typical sampling/distribution heavy simulation. I'd suggest that you put a C-enum in the header then use our fancy MooseEnum to enum converter and use a much faster switch statement here.